### PR TITLE
[design doc] new outline for documentation

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <krisss@vmware.com>

Here's a straw-man proposal for a revised outline for our documentation. I drew inspiration from https://istio.io/docs and https://kubernetes.io/docs.

I'd like to:
- agree on the outline (sections and pages)
- do whatever reorganization we can now (without rewriting)
- add issues to track specific page writes/rewrites

Please comment away! cc @carlisia @prydonius

### CODE REVIEW

File: design/docs-reorg.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+- Concepts
+    - Overview
+        - Core Features
+        - Architecture
+    - Storage Providers/Plugins
+    - Backup Storage Locations
+    - Volume Snapshot Locations
+    - Restic Integration
+- Setup
```
Comment (skriss) [inline design/docs-reorg.md]: somewhere in this section I'd like to explain that `velero install` will typically set up one BSL and optionally one VSL, but that you can configure more of these afterwards with `velero backup-location create` / `velero snapshot-location create`.

File: design/docs-reorg.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+The Velero documentation has evolved organically since the project's inception.
+While it contains a lot of useful information, it is not currently organized in a way that is easy to navigate for users or contributors.
+For example, the introductory section jumps straight into a discussion of Velero's architecture, rather than starting with a good overview of its core features.
+Also, the documentation on how to actually *use* Velero is fairly superficial, only covering the most straight-forward use cases.
+Additionally, different types of information are often mixed together (i.e. conceptual overviews along with detailed setup instructions).
+
+This design document proposes a new outline for the documentation that is intended to be more easily navigated by both users and contributors.
+The new outline both reorganizes the existing content, *and* calls for new content where relevant.
+
+## High-Level Design
```
Comment (ashish-amarnath) [inline design/docs-reorg.md]: Who will be the target audience of the doc that will be created? That would be a good indicator of how we want to organize this.

Comment (skriss) [reply design/docs-reorg.md]: IMHO the main audience is users. There's also a "core contributors" audience and a "plugin developers" audience. I think that covers it but LMK if you think there are others!

Comment (ashish-amarnath) [reply design/docs-reorg.md]: So this is what I am thinking. We can have like a Quick Start doc that folks would use to kick the tires and get see velero in action. This is where we would cover velero installation, backup store setup and volume snapshotter setup for public clouds. Then there can be advanced user docs that would, maybe, a deep dive into some scenarios.
The plugin developers doc would be a separate doc.

File: design/docs-reorg.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    - Install sample workload
+    - Take a backup
+    - Create a scheduled backup
+    - Recover a namespace
+    - Recover from full-cluster failure
+    - Clone a namespace
+    - Migrate to a new cluster
+    - Run velero in a different namespace
+    - Configure a hook
+    ...
```
Comment (kmova) [inline design/docs-reorg.md]: Should there be an sub-section here that could probably give details on a specific application? More on the lines of 

```
- Example Application Hooks
   - PostgreSQL 
   - Percona
   - Minio
```

**Comment by carlisia** [comment]
Here's one issue to track to write and add to the documentation: https://github.com/vmware-tanzu/velero/issues/1440

File: design/docs-reorg.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    - ZenHub
+    - Development
+    - Build from source
+    - Run locally
+    - Plugin development
+- Reference
+    - Backup file format
+    - API types reference
+
+To be determined:
```
Comment (carlisia) [inline design/docs-reorg.md]: I like having both of these subjections. Hopefully we'll find a way to collapse sub-items as these can also run very long.

**Comment by carlisia** [comment]
An issue for Q&A content: https://github.com/vmware-tanzu/velero/issues/1809

File: design/docs-reorg.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+- FAQ?
+- Troubleshooting?
```
Comment (nrb) [inline design/docs-reorg.md]: I would really like to keep troubleshooting around, as it's a useful place to hold information for helping users help themselves.

File: design/docs-reorg.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+This design document proposes a new outline for the documentation that is intended to be more easily navigated by both users and contributors.
+The new outline both reorganizes the existing content, *and* calls for new content where relevant.
+
+## High-Level Design
+
+- Concepts
+    - Overview
+        - Core Features
+        - Architecture
+    - Storage Providers/Plugins
```
Comment (nrb) [inline design/docs-reorg.md]: I might split these up, or perhaps add a new page - while the storage providers have been the most important plugins usually, there are also the ItemAction plugins.

That plugin page that I'm thinking of may, however, be out of scope for this particular redesign, as I envision it to be something that discusses the types of plugins that Velero supports, their APIs, and possibly how to write one.

File: design/docs-reorg.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            - Resource requirements
+        - Download the Velero release
+        - Install on your platform
+    - Supported Storage Providers
+    - Evaluation Install (using in-cluster MinIO)
+    - Restic setup
+    - Uninstalling
+    - `velero install` reference
+        - cover some of the most common flags
+- Use (TODO add some subsections for organization)
```
[the remark this answers is not in the document]
Comment (skriss) [reply design/docs-reorg.md]: Yeah, I would really like to get collapsible sections added to the site, agree this is a problem.

**Comment by skriss** [comment]
I'm going to close this out as it's stale, we can pick it up if/when we do work here.

**Pull request closed without merging**